### PR TITLE
Improve process for committing changes to the license information files

### PR DESCRIPTION
**Pull request by aosoriodw**
The GitHub action that commits changes to license information after a dependabot change uses the GitHub token for the job. However, this doesn't work well, because this token does not force a re-run of the license verification job, and there is no way to re-run the job without manual intervention.
This PR changes the following:

1. Is uses a personal token to commit changes, which will trigger a re-run of the license workflow after a commit by the action.
2. It checks that go.mod and go.sum are clean before generating license information. This is necessary because the license files should be generated from the information committed to GitHub
3. Verified that the action didn't to any changes to go.mod and go.sum, for the same reason explained in # 2.

### CODE REVIEW

**Review by joseldmc — APPROVED** [review]
LGTM, perhaps I would like at least someone else (@arturogonzalez58 or @AidanWenzel ) to review the PR.

File: cmd/go-mkopensource/main.go [hunk 1]
```diff
@@ -430,9 +436,56 @@ func Main(args *CLIArgs) error {
 		}
 	}
 
+	isDirty, err := isGoModDirty()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "WARNING: could not verify if go.mod or go.sum are dirty: %s.\n", err.Error())
+	}
+
+	if isDirty {
+		fmt.Fprintln(os.Stderr, "WARNING: go.mod or go.sum are dirty.\nMake sure that these files are commited with the "+
```
Comment (AidanWenzel) [inline cmd/go-mkopensource/main.go:445]: Should the workflow fail if this happens? Same for if `isGoModDirty` errs.

Comment (aosoriodw) [reply cmd/go-mkopensource/main.go]: Good catch! I've updated it.

Comment (aosoriodw) [reply cmd/go-mkopensource/main.go]: For the `err != nil` it's just a warning because we can't be sure the files were modified.

**Pull request merged**